addons/project/models/project_task.py

```diff
@@ -103,7 +103,7 @@ def _default_company_id(self):
     @api.model
     def _read_group_stage_ids(self, stages, domain, order):
         search_domain = [('id', 'in', stages.ids)]
-        if 'default_project_id' in self.env.context:
+        if 'default_project_id' in self.env.context and 'subtask_action' not in self._context:
```

Comment:
```suggestion
        if 'default_project_id' in self.env.context and not self._context.get('subtask_action'):
```

subtask_action could be in the context and the value could be falsy as well. 🙂